internal/app/cache/cache.go

```diff
@@ -0,0 +1,31 @@
+package cache
+
+import (
+	envoy_service_discovery_v3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+)
+
+type Cache interface {
+	// Exists returns true if the key exists in the cache, false otherwise.
+	Exists(key string) bool
+
+	// Fetch returns the cached response if it exists.
+	Fetch(key string) (*envoy_service_discovery_v3.DiscoveryResponse, error)
+
+	// IsStreamOpen returns false if no stream is opened against the upstream control plane for the given key.
+	IsStreamOpen(key string) (bool, error)
+
+	// SetStreamOpen sets whether there is a stream open against the upstream control plane for the given key.
+	SetStreamOpen(key string, open bool) error
```

Comment:
Because the setting of the stream seems to always be tied to other actions (setting response, adding watch), I don't think this method is needed in the interface.